src/peergos/shared/user/UserUtil.java

```diff
@@ -22,7 +22,7 @@
                                                                SecretGenerationAlgorithm algorithm) {
         if (password.equals(username))
             return Futures.errored(new IllegalStateException("Your password cannot be the same as your username!"));
-        CompletableFuture<byte[]> fut = hasher.hashToKeyBytes(username, password, algorithm);
+        CompletableFuture<byte[]> fut = hasher.hashToKeyBytes(username + algorithm.getExtraSalt(), password, algorithm);
```

Comment:
For my own understanding, why add the salt to the username and not the password?

Reply:
The username is the salt. We are just extending it.

Reply:
Ah I see.